compiler/rustc_borrowck/src/session_diagnostics.rs

```diff
@@ -307,63 +336,150 @@ pub(crate) struct LifetimeMismatchOpaqueParam<'tcx> {
     pub prev_span: Span,
 }
 
-#[derive(Subdiagnostic)]
 pub(crate) enum CaptureReasonLabel<'a> {
-    #[label(borrowck_moved_due_to_call)]
     Call {
-        #[primary_span]
         fn_call_span: Span,
-        place_name: &'a str,
-        is_partial: bool,
+        place_name_call: &'a str,
+        is_partial_call: bool,
         is_loop_message: bool,
     },
-    #[label(borrowck_moved_due_to_usage_in_operator)]
     OperatorUse {
-        #[primary_span]
         fn_call_span: Span,
-        place_name: &'a str,
-        is_partial: bool,
+        place_name_use: &'a str,
+        is_partial_use: bool,
         is_loop_message: bool,
     },
-    #[label(borrowck_moved_due_to_implicit_into_iter_call)]
     ImplicitCall {
-        #[primary_span]
         fn_call_span: Span,
-        place_name: &'a str,
-        is_partial: bool,
+        place_name_implicit_call: &'a str,
+        is_partial_implicit_call: bool,
         is_loop_message: bool,
     },
-    #[label(borrowck_moved_due_to_method_call)]
     MethodCall {
-        #[primary_span]
         fn_call_span: Span,
-        place_name: &'a str,
-        is_partial: bool,
+        place_name_method_call: &'a str,
+        is_partial_method_call: bool,
```

Comment:
These renames are unfortunate. We do want these to be fairly independently nameable. These are just there because we're not guaranteed the main diagnostic will have these values set, right? The fields are effectively redundant, so we could

* remove them (and rely on the main diagnostic to set them)
* change the assert to only prevent overwriting args if the value is different

Thoughts?

Reply:
> I think the dependencies should follow a tree structure, where a subdiagnostic can read the field of the main diagnostic, but the field of a subdiagnostic cannot be read by either the diagnostic or another subdiagnostic.

Agreed